dallinger/mturk.py

```diff
@@ -352,62 +461,96 @@ def grant_bonus(self, assignment_id, amount, reason):
+             'AWSMechanicalTurkDataSchemas/2006-07-14/ExternalQuestion.xsd">'
+             '<ExternalURL>{}</ExternalURL>'
+             '<FrameHeight>{}</FrameHeight></ExternalQuestion>')
+        return q.format(url, frame_height)
+
+    def _request_token(self):
+        return str(time.time())
+
     def _translate_hit(self, hit):
         translated = {
-            'id': hit.HITId,
-            'type_id': hit.HITTypeId,
-            'created': timestr_to_dt(hit.CreationTime),
-            'expiration': timestr_to_dt(hit.Expiration),
-            'max_assignments': int(hit.MaxAssignments),
-            'title': hit.Title,
-            'description': hit.Description,
-            'keywords': hit.Keywords.split(', '),
-            'qualification': getattr(hit, 'QualificationTypeId', None),
-            'reward': float(hit.Amount),
-            'review_status': hit.HITReviewStatus,
-            'status': hit.HITStatus,
+            'id': hit['HITId'],
+            'type_id': hit['HITTypeId'],
+            'created': hit['CreationTime'],
+            'expiration': hit['Expiration'],
+            'max_assignments': hit['MaxAssignments'],
+            'title': hit['Title'],
+            'description': hit['Description'],
+            'keywords': [w.strip() for w in hit['Keywords'].split(',')],
+            'qualification_type_ids': [
+                q['QualificationTypeId'] for q in hit['QualificationRequirements']
+            ],
+            'reward': float(hit['Reward']),
+            'review_status': hit['HITReviewStatus'],
+            'status': hit['HITStatus'],
         }
 
         return translated
 
     def _translate_qtype(self, qtype):
         return {
-            'id': qtype.QualificationTypeId,
-            'created': timestr_to_dt(qtype.CreationTime),
-            'name': qtype.Name,
-            'description': qtype.Description,
-            'status': qtype.QualificationTypeStatus,
+            'id': qtype['QualificationTypeId'],
+            'created': qtype['CreationTime'],
+            'name': qtype['Name'],
+            'description': qtype['Description'],
+            'status': qtype['QualificationTypeStatus'],
         }
 
-    def _is_ok(self, mturk_response):
-        return mturk_response == []
-
-    def approve_assignment(self, assignment_id):
-        try:
-            self.mturk.approve_assignment(assignment_id, feedback=None)
-        except MTurkRequestError:
-            raise MTurkServiceException(
-                "Failed to approve assignment {}".format(assignment_id))
-        return True
-
-    def expire_hit(self, hit_id):
-        try:
-            self.mturk.expire_hit(hit_id)
-        except MTurkRequestError:
-            raise MTurkServiceException(
-                "Failed to expire HIT {}".format(hit_id))
-        return True
+    def _is_ok(self, response):
+        return response == {} or response.keys() == ['ResponseMetadata']
```

Comment:
`_is_ok` seems like a confusing name for a function that looks like it's actually checking if the response is empty